clouddriver-saga/src/main/kotlin/com/netflix/spinnaker/clouddriver/saga/events.kt

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.clouddriver.saga
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.fasterxml.jackson.annotation.JsonTypeName
+import com.netflix.spinnaker.clouddriver.event.SpinnakerEvent
+import com.netflix.spinnaker.clouddriver.saga.models.Saga
+import com.netflix.spinnaker.kork.exceptions.SpinnakerException
+
+/**
+ * Root event type for [Saga]s.
+ *
+ * @property sagaName Alias for [aggregateType]
+ * @property sagaId Alias for [aggregateId]
+ */
+abstract class SagaEvent(
+  sagaName: String,
+  sagaId: String
+) : SpinnakerEvent(sagaName, sagaId) {
+  val sagaName
+    @JsonIgnore get() = aggregateType
+
+  val sagaId
+    @JsonIgnore get() = aggregateId
+}
+
+/**
+ * Emitted whenever a [Saga] is saved.
+ *
+ * This event does not attempt to find a difference in state, trading off persistence verbosity for a little bit
+ * of a simpler implementation.
+ *
+ * @param saga The [Saga]'s newly saved state
+ */
+@JsonTypeName("sagaSaved")
+class SagaSaved(
+  val saga: Saga
+) : SagaEvent(saga.name, saga.id)
+
+/**
+ * Emitted whenever an internal error has occurred while applying a [Saga].
+ *
+ * @param reason A human-readable cause for the error
+ * @param error The Exception (if any) that caused the error condition
+ * @param retryable Flags whether or not this error is recoverable
+ * @param data Additional data that can help with diagnostics of the error
+ */
+@JsonTypeName("sagaInternalErrorOccurred")
+class SagaInternalErrorOccurred(
+  sagaName: String,
+  sagaId: String,
+  val reason: String,
+  val error: Exception? = null,
+  val retryable: Boolean = true,
+  val data: Map<String, String> = mapOf()
```

Comment:
I like this standard functionality to provide nice errors and things that may be helpful.